src/main.cpp

```diff
@@ -16,20 +16,30 @@ int main(int argc, char *argv[]) {
 		return 0;
 	}
 
-	for (int i = 1; i < argc; ++i) {
-		if (file_included(argv[i])) {
-			filename = argv[i];
-		} else if (argv[i] == std::string("-o")) {
-			frontend.outputFileName = std::string(argv[++i]);
+	try {
+		if (argv[argc - 1] == std::string("-h")) {
+			Frontend::Helper(helpArgument);
+			return 0;
 		}
+		for (int i = 1; i < argc; ++i) {
+			if (file_included(argv[i])) {
+				filename = argv[i];
+			} else if (argv[i] == std::string("-o")) {
+				frontend.outputFileName = std::string(argv[++i]);
+			} else {
+				throw std::invalid_argument(argv[i]);
+			}
+		}
+		if (drv.parse_file(filename) == 0) {
+			frontend.drv = &drv;
+			frontend.WriteFile();
+		} else {
+			std::cout << "Compilation error" << std::endl;
+			Frontend::Helper();
+		}
+		return 0;
+	} catch (std::invalid_argument &ex) {
```

Comment:
Jeg tror ikke at try-catches er den rigtige måde at håndtere det her på. At wrappe alt koden i en try-catch, især dele af koden der ikke kan kaste den exception, er ret grimt, når man bare kunne skrive print i stedet for `throw`. 

Desuden, nu hvor vi er i gang med at fixe `main`, burde vi smide et `return 1` til sidst. Derefter kan vi explicit returne 0 hvis at vores compilation faktisk lykkedes. Lige nu returner vi f.eks. stadig 0 under en compilation error, hvilket jo er helt galt.